include/internal/catch_console_colour.cpp

```diff
@@ -169,7 +169,12 @@ namespace {
 #ifdef CATCH_PLATFORM_MAC
             !isDebuggerActive() &&
 #endif
-            isatty(STDOUT_FILENO);
+#if !defined(__DJGPP__) || !defined(__STRICT_ANSI__)
```

Comment:
Yes, g++ defines `__STRICT_ANSI__` whenever supplied with `-std=c++..` but not when supplied with `-std=gnu++..` which can be controlled with CMake's `CXX_EXTENSIONS`. It can also be forced using `-ansi` flag.
Anyhow, this condition should guarantee that `isatty` is **not** used only if `__DJGPP__ && __STRICT_ANSI__`.